### PR TITLE
Update Calcite*Test to use junit5

### PR DESCRIPTION
**Pull request by kgyrtkirk**
* change the way temp dirs are handled
* add openrewrite workflow to safeguard upgrade; possibly make future changes easier
* replace junitparamrunner with standard junit5 parametered tests
* make `QueryLogHook` less complicated
* update a few rules to junit5 api
* lots of boring changes `@Before` / `@BeforeEach` and alike
* remove `@RunWith(Nested.class)` and [deconfigure the default surefire exclusion for nested static classes](https://github.com/junit-team/junit5/issues/1377)

Fixes #XXXX.

#### Release note

<hr>

##### Key changed/added classes in this PR
 * `MyFoo`
 * `OurBar`
 * `TheirBaz`

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by kgyrtkirk** [comment]
Right now openrewrite is configured to upgrade the `Calcite*Test` tests to junit5; if we decide to move on upgrading more - we can first change its configuration - let it do the bulk of it.

Having it enable and run on every build will make sure deprecated/etc stuff is not being pulled back in into these files.

The failing check is there because I've configured an extra rule which should be undone....I'll remove it when I'll update the branch next time

**Comment by gianm** [comment]
Ah, I see the `rewrite.yml` with the config. Seems neat. I see that the static failure is about too many `UnnecessaryThrows`, maybe we should remove that rule?

**Comment by kgyrtkirk** [comment]
there is something with the `CalciteSelectJoinQueryMSQTest` ; it did run locally for me - but not by the CI.
let's see if `@Nested` helps

File: pom.xml [hunk 1]
```diff
@@ -1741,6 +1768,9 @@
                         <!-- our tests are very verbose, let's keep the volume down -->
                         <redirectTestOutputToFile>true</redirectTestOutputToFile>
                         <forkNode implementation="org.apache.maven.plugin.surefire.extensions.SurefireForkNodeFactory"/>
+                        <excludes>
```
Comment (gianm) [inline pom.xml:1771]: Please include some comment about why we are doing this. I think it's about removing a default exclusion for static inner classes, but it is not obvious.

**Review by gianm — COMMENTED** [review]
I skimmed the automated changes and tried to focus review on the ones that appeared to have been done by hand. They look OK to me. Just had some questions (see next comment) and one comment about the new surefire config in `pom.xml`.

**Comment by gianm** [comment]
@kgyrtkirk,

> I see that the static failure is about too many UnnecessaryThrows, maybe we should remove that rule?

How was this resolved? I don't see it failing anymore.

> there is something with the `CalciteSelectJoinQueryMSQTest` ; it did run locally for me - but not by the CI.

Did it help?

**Comment by kgyrtkirk** [comment]
> >  UnnecessaryThrows
> How was this resolved? I don't see it failing anymore.

I've removed the global enablement of `UnnecessaryThrows` ; we can do that later - I was over-zealous adding it right next to pushing the branch - it did not fail on the few modules I was checking :)

> > there is something with the CalciteSelectJoinQueryMSQTest ; it did run locally for me - but not by the CI.
> Did it help?

my investigation have lead to uncover that `Surefire` has that [odd exclusion setting](https://github.com/junit-team/junit5/issues/1377) - removing all `@RunWith(Nested.class)` and disabling the exclusion should be much better

**Review by gianm — APPROVED** [review]
LGTM

**Pull request merged**